### PR TITLE
DLS-10422 Use Play Actions instead of functions to authenticate

### PR DESCRIPTION
**Pull request by LAJW**

### CODE REVIEW

File: app/uk/gov/hmrc/cbcr/package.scala [hunk 1]
```diff
@@ -51,4 +52,10 @@ package object cbcr {
     l.map(t => t._1 -> toJsFieldJsValueWrapper(t._2))
   implicit def listJsWrapper[T: Writes](l: List[T]): List[JsValueWrapper] = l.map(toJsFieldJsValueWrapper[T])
 
+  implicit class KibanaLogger(logger: _root_.play.api.Logger) {
```
Comment (johno1985) [inline app/uk/gov/hmrc/cbcr/package.scala:55]: what is this for?

File: test/uk/gov/hmrc/cbcr/controllers/ReportingEntityDataControllerSpec.scala [hunk 2]
```diff
@@ -110,7 +115,7 @@ class ReportingEntityDataControllerSpec extends UnitSpec with ScalaFutures with
     }
 
     "respond with a 400 if ReportingEntity in request is invalid" in {
-      val result = controller.save()(badFakePostRequest)
+      val result = controller.save()(badFakePostRequest).run()
```
Comment (johno1985) [inline test/uk/gov/hmrc/cbcr/controllers/ReportingEntityDataControllerSpec.scala:118]: what does .run() do here?

**Review by johno1985 — COMMENTED** [review]
couple of minor comments/questions

File: app/uk/gov/hmrc/cbcr/package.scala (hunk 1 above)
Comment (LAJW) [reply app/uk/gov/hmrc/cbcr/package.scala:55]: This is a part of the other PR. Good spot, should be removed.

File: test/uk/gov/hmrc/cbcr/controllers/ReportingEntityDataControllerSpec.scala (hunk 2 above)
Comment (LAJW) [reply test/uk/gov/hmrc/cbcr/controllers/ReportingEntityDataControllerSpec.scala:118]: Typed Actions when provided with a body of a wrong type return an "accumulator", which needs to be run in order to return the Response.

File: app/uk/gov/hmrc/cbcr/package.scala (hunk 1 above)
Comment (LAJW) [reply app/uk/gov/hmrc/cbcr/package.scala]: Removed.

**Review by johno1985 — APPROVED** [review]
Lets roll this out elsewhere

**Comment by TomRafferty** [comment]
was some work done on a different machine or something which lacks the commit signing?

**Pull request merged**